### PR TITLE
feat: add learning base url

### PR DESCRIPTION
**Pull request by asadali145**
### What are the relevant tickets?

Fixes https://github.com/mitodl/hq/issues/3475

### Description (What does it do?)

- Adds a new config `LEARNING_BASE_URL`.
- Adds it for all the apps and all environments.

### How can this be tested?

Add it to the .env.development for the authoring MFE and test it.

### CODE REVIEW

File: src/ol_concourse/pipelines/open_edx/mfe/pipeline.py [hunk 1]
```diff
@@ -84,6 +85,7 @@ def mfe_params(
             f"{open_edx.environment}-open-edx-language-preference"
         ),
         "LMS_BASE_URL": f"https://{open_edx.lms_domain}",
+        "LEARNING_BASE_URL": f"https://{open_edx.learning_base_url}",
```
Comment (arslanashraf7) [inline src/ol_concourse/pipelines/open_edx/mfe/pipeline.py:88]: I think we can make this change more robust like. This way we would not need to add this to `src/ol_concourse/pipelines/open_edx/mfe/values.py`. So, the only change would be this one line.

```suggestion
        "LEARNING_BASE_URL": f"https://{open_edx.learning_base_url}/{mfe.application.path}",
```
OR

```suggestion
        "LEARNING_BASE_URL": f"https://{open_edx.learning_base_url}/learn",
```

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
I requested some changes, But it would be best to get opinion from @mitodl/devops as well.

File: src/ol_concourse/pipelines/open_edx/mfe/pipeline.py (hunk 1 above)
Comment (asadali145) [reply src/ol_concourse/pipelines/open_edx/mfe/pipeline.py:88]: Yeah, this 2nd suggestion seems best. I am updating it.

Comment (asadali145) [reply src/ol_concourse/pipelines/open_edx/mfe/pipeline.py]: I am not sure about a way to get the learning MFE path dynamically here. @mitodl/devops Could suggest one maybe?

Comment (blarghmatey) [reply src/ol_concourse/pipelines/open_edx/mfe/pipeline.py]: I just pushed a commit that will automatically pull in the proper value. I started to write out how to do it, but it was easier to just show you with code :)

Comment (asadali145) [reply src/ol_concourse/pipelines/open_edx/mfe/pipeline.py]: Thanks, This makes much more sense now.

**Review by arslanashraf7 — APPROVED** [review]

**Pull request merged**